src/components/ScholarshipAmountField.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import {
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  TextField,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import AmountType from '../types/AmountType';
+
+const useStyles = makeStyles({
+  formControl: {
+    minWidth: 200,
+  },
+});
+
+function ScholarshipAmountField(props) {
+  const classes = useStyles();
+  const {
+    amountType,
+    minAmount,
+    maxAmount,
+    onTypeChange,
+    onMinChange,
+    onMaxChange,
+    updateMinMax,
+  } = props;
+
+  function displayAmountFields() {
+    if (amountType === AmountType.Fixed) {
+      return (
+        <TextField
+          label="Amount"
+          type="number"
+          InputLabelProps={{ shrink: true }}
+          fullWidth
+          InputProps={{ inputProps: { min: 0, step: 50 } }}
+          value={minAmount}
+          onChange={(e) => updateMinMax(parseInt(e.target.value, 10) || '')}
```

Comment:
Would it perhaps be useful to refactor a separate `AmountTextField` component that has most of these properties set? It could do the integer parsing as well and set e.target.value to a number or `''`